Libraries/libbom/Tools/lsbom.cpp

```diff
@@ -294,6 +294,81 @@ extern "C" {
     } __attribute__((packed));
 }
 
+static std::string
+PrintItemToString(Options::PrintItem item, std::string path, struct bom_path_info_2 *info)
+{
+    switch (item) {
+        case Options::PrintItem::Checksum:
+            if (info->type == bom_path_type_directory) {
+                return "";
+            }
+            return std::to_string(ntohl(info->checksum));
+        case Options::PrintItem::FileName:
+            return path;
+        case Options::PrintItem::FileNameQuoted:
+            return "\"" + path + "\"";
+        case Options::PrintItem::GroupID:
+            return std::to_string(ntohl(info->group));
+        case Options::PrintItem::GroupName:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::Permissions: {
+            std::stringstream stream;
+            stream << std::oct << ntohs(info->mode);
+            return stream.str();
+            }
+        case Options::PrintItem::PermissionsText:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::FileSize:
+            if (info->type == bom_path_type_directory) {
+                return "";
+            }
+            return std::to_string(ntohl(info->size));
+        case Options::PrintItem::FileSizeFormatted:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::ModificationTime:
+            return std::to_string(ntohl(info->modtime));
+        case Options::PrintItem::ModificationTimeFormatted:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::UserID:
+            return std::to_string(ntohl(info->user));
+        case Options::PrintItem::UserName:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::UserGroupID:
+            return std::to_string(ntohl(info->user)) + "/" + std::to_string(ntohl(info->group));
+        case Options::PrintItem::UserGroupName:
+            return std::string(); // TODO: implement
+    }
+}
+
+static std::vector<Options::PrintItem>
+GetDefaultPrintFormat ()
+{
+    std::vector<Options::PrintItem> toReturn;
+
+    toReturn.push_back(Options::PrintItem::FileName);
+    toReturn.push_back(Options::PrintItem::Permissions);
+    toReturn.push_back(Options::PrintItem::UserGroupID);
+    toReturn.push_back(Options::PrintItem::FileSize);
+    toReturn.push_back(Options::PrintItem::Checksum);
+
+    return toReturn;
```

Comment:
The initialiser list looks like a slightly faster, cleaner way to do this.
I'll make that change, thanks!